protocols/imap/src/main/java/org/apache/james/imap/processor/base/SelectedMailboxImpl.java

```diff
@@ -75,7 +76,7 @@
     private boolean isDeletedByOtherSession = false;
     private boolean sizeChanged = false;
     private boolean silentFlagChanges = false;
-    private final Flags applicableFlags;
+    private final AtomicReference<Flags> applicableFlags;
```

Comment:
Is there any case where we use `SelectedMailboxImpl` concurrently?

Reply:
We might read applicableFlags and update it concurrently while receiving an email, so yes, we need to handle concurrency.

Reply:
My question was more: does it have to be completely thread-safe or do we know the exact concurrency cases that happen? Because if any concurrent call can happen we have to fix it properly IMO

Reply:
`event` method can be called concurrently to over methods.

Over methods cannot be called concurrently as they are executed within the same TCP session.